src/main/java/Floda.java

```diff
@@ -0,0 +1,248 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.FileNotFoundException;
+
+
+public class Floda {
+    private static ArrayList<Task> list = new ArrayList<>();
+    private final static String NAME = "Floda";
+    private static final String FILE_PATH = "./data/tasks.txt";
+
+
+    public static void main(String[] args) {
+        System.out.println("Hello! I'm " + NAME);
+        try {
+            checkIfFileExists();
+            readFromFile();
+        } catch (FileNotFoundException | InvalidInputException e) {
+            System.out.println("Error: " + e.getMessage());
+        }
+        Scanner scanner = new Scanner(System.in);
+        System.out.println("I can keep track of a to-do list for you! Just type what you want to add to the list.");
+        while (true) {
+            try {
+                String line = scanner.nextLine().trim();
+                String[] parts = line.split(" ", 2);
+                String command = parts[0].trim();
+                switch (command) {
+                    case "bye":
+                        System.out.println("Bye. Hope to see you again soon!");
+                        scanner.close();
+                        saveToFile();
+                        return;
+                    case "list":
+                        if (list.isEmpty()) {
+                            System.out.println("Your to-do list is empty.");
+                        } else {
+                            System.out.println("List so far: ");
+                            for (int i = 0; i < list.size(); i++) {
+                                System.out.println((i + 1) + "." + list.get(i));
+                            }
+                        }
```

Comment:
The switch case is a good application of SLAP. I think you can further improve on SLAP on the highlighted section (bye and list)!